libs/zephyr-agent/src/lib/auth/login.ts

```diff
@@ -43,30 +43,33 @@ export async function checkAuth(): Promise<string> {
     await removeToken();
   }
 
+  // In non-TTY environments it's expected that a ZE_SECRET_TOKEN is present
+  // since user cannot interact with it.
+  if (!isTTY) {
+    logFn('warn', `Could not load ${StorageKeys.ze_secret_token}.`);
+  }
+
   // No valid token found; initiate authentication.
   logFn('', `${yellow('Authentication required')} - You need to log in to Zephyr Cloud`);
 
   // Get authentication URL first
   const sessionKey = generateSessionKey();
   const authUrl = await getAuthenticationURL(sessionKey);
 
-  // Prompt user to continue
-  await promptForAuthAction(authUrl);
+  const browserController = new AbortController();
 
-  // Handle browser opening
-  try {
-    await openUrl(authUrl);
-    logFn('', `${blue('⏳')} Waiting for authentication...\n`);
-  } catch (error) {
-    // If browser failed to open, fall back to manual
-    fallbackManualLogin(authUrl);
-  }
+  // Tries to open the browser to authenticate the user
+  void promptForAuthAction(authUrl, browserController.signal)
```

Comment:
IMHO prompt should lead to a race promise and not use exception in this way (cuz exception is unexpected behaviour, like no internet and socket dropped off )

Reply:
@valorkin, at first, you may think that this can be raced between both, but in reality, we just need to wait for the auth confirmation, any other kind of prompt should be done async, and there's no constraint requiring the browser to be opened to have a valid login.

The catch handler is not for the first promise, but for the second one in case our runtime `eval` of `open` module fails.

https://github.com/ZephyrCloudIO/zephyr-packages/blob/469c0751a0689e3901a3c4e73663e5933cedc162/libs/zephyr-agent/src/lib/auth/login.ts#L136-L140

Putting it at last just adds a safety net in case something unexpected happens with `readline.question`

Reply:
Why manual login is in catch?